File: webflux-test/src/main/java/com/example/dto/UserResponse.java
package com.example.dto;

import com.example.repository.User;
import lombok.Builder;

import java.time.LocalDateTime;

public record UserResponse(
        Long id,
        String name,
        String email,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {

    @Builder
    public UserResponse {}

    public static UserResponse of(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .name(user.getName())
                .email(user.getEmail())
                .createdAt(user.getCreatedAt())
                .updatedAt(user.getUpdatedAt())
                .build();
    }
}

File: library-app/src/main/kotlin/com/example/libraryapp/dto/user/response/UserResponse.kt
package com.example.libraryapp.dto.user.response

import com.example.libraryapp.domain.user.User

data class UserResponse(
    val id: Long,
    val name: String,
    val age: Int?
) {

    companion object {
        fun of(user: User): UserResponse {
            return UserResponse(
                id = user.id!!,
                name = user.name,
                age = user.age
            )
        }
    }
}